libpcap/compile.go

```diff
@@ -0,0 +1,202 @@
+package libpcap
+
+import (
+	"fmt"
+	"unsafe"
+
+	"github.com/cilium/ebpf/asm"
+	"github.com/cloudflare/cbpfc"
+	"golang.org/x/net/bpf"
+)
+
+/*
+#cgo LDFLAGS: -L/usr/local/lib -lpcap -static
+#include <stdlib.h>
+#include <pcap.h>
+*/
+import "C"
+
+type pcapBpfProgram C.struct_bpf_program
+
+const (
+	MaxBpfInstructions       = 4096
```

Comment:
You're right, but this is for compiling pcap-filter to cbpf, which is concise and I consider 4096 is sufficient so that we don't have to waste unnecessary memory.

For instance, we can implement equivalent filter to `pwru --filter-src-ip --filter-dst-ip --filter-src-port --filter-dst-port --filter proto --filter-tcp-flag syn` using only 19 pieces of cbpf instructions:

```
$ tcpdump -d 'src host 1.1.1.1 and dst host 2.2.2.2 and src port 23 and dst port 23 and tcp and tcp[tcpfl
ags] & tcp-syn != 0'
Warning: assuming Ethernet
(000) ldh      [12]
(001) jeq      #0x800           jt 2	jf 19
(002) ld       [26]
(003) jeq      #0x1010101       jt 4	jf 19
(004) ld       [30]
(005) jeq      #0x2020202       jt 6	jf 19
(006) ldb      [23]
(007) jeq      #0x84            jt 19	jf 8
(008) jeq      #0x6             jt 9	jf 19
(009) ldh      [20]
(010) jset     #0x1fff          jt 19	jf 11
(011) ldxb     4*([14]&0xf)
(012) ldh      [x + 14]
(013) jeq      #0x17            jt 14	jf 19
(014) ldh      [x + 16]
(015) jeq      #0x17            jt 16	jf 19
(016) ldb      [x + 27]
(017) jset     #0x2             jt 18	jf 19
(018) ret      #262144
(019) ret      #0
```